test/clj/chronograph/domain/timer_test.clj

```diff
@@ -0,0 +1,480 @@
+                                                              task-id
+                                                              "A note by an admin.")]
+                                     (timer/start! db/datasource
+                                                   admin-id
+                                                   (:timers/id timer))
+                                     timer)
+            running-timer-by-member (let [timer (timer/create! db/datasource
+                                                               member-id
+                                                               task-id
+                                                               "A note by an admin.")]
+                                      (timer/start! db/datasource
+                                                    member-id
+                                                    (:timers/id timer))
+                                      timer)]
+        ;; first ensure running timers remain running
+        (is (nil? (timer/stop! db/datasource
+                               member-id
+                               (:timers/id running-timer-by-admin)))
+            "A member cannot stop an admin's timer.")
+        (is (nil? (timer/stop! db/datasource
+                               admin-id
+                               (:timers/id running-timer-by-member)))
+            "An admin cannot start a member's timer.")
+        ;; now double-check that each can stop their respective timers
+        (is (s/valid? :time-spans/time-span
+                      (timer/stop! db/datasource
+                                   member-id
+                                   (:timers/id running-timer-by-member)))
+            "A member can stop their own timer.")
+        (is (s/valid? :time-spans/time-span
+                      (timer/stop! db/datasource
+                                   admin-id
+                                   (:timers/id running-timer-by-admin)))
+            "An admin can stop their own timer.")))))
+
+
+;; ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;; Find Timer Tests
+;; ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+
+
+(deftest find-timer-by-timer-id
+  (testing "when we find a timer by timer id, and it was started and/or stopped at least once"
+    (let [member-id (:member-id @test-context)
+          {timer-id :timers/id} (timer/create! db/datasource
+                                               member-id
+                                               (:task-id-1 @test-context)
+                                               "A sample note.")]
+      ;; start/stop the timer a few times
+      (dotimes [_ 3]
+        (timer/start! db/datasource
+                      member-id
+                      timer-id)
+        (timer/stop! db/datasource
+                     member-id
+                     timer-id))
+      ;; validate timer data retrieved
+      (is (s/valid? :domain.timer/find-by-id-retval
```

Comment:
It'll be valuable to check at least the number of timespans returned.

Reply:
By freezing time, we should also be able to assert on the values of the timespans.

Reply:
Done... froze time and comparing created input values to retrieved values.